pkg/webhook/genericsimpleevent.go

```diff
@@ -55,57 +50,46 @@ func (g *genericWebhook) Handle(c *gin.Context) {
 		return
 	}
 
-	gwData := &genericWebhookData{}
-
-	err = c.BindJSON(gwData)
+	payload, err := ioutil.ReadAll(c.Request.Body)
 	if err != nil {
-		log.Printf("Failed to convert POST data into JSON: %s", err)
-		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed POST data"})
+		log.Printf("Failed to read body: %s", err)
+		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed body"})
 		return
 	}
+	defer c.Request.Body.Close()
 
-	body, err := ioutil.ReadAll(c.Request.Body)
-
+	// try to unmarshal Revision data, if they do exist
+	revision := &brigade.Revision{}
+	err = json.Unmarshal(payload, &revision)
 	if err != nil {
-		log.Printf("Failed to read body: %s", err)
-		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed body"})
+		log.Printf("Failed to convert POST data into JSON: %s", err)
+		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed POST data - Invalid JSON"})
 		return
 	}
-	defer c.Request.Body.Close()
 
-	go g.notifyGenericWebhookEvent(proj, body, gwData)
-	c.JSON(200, gin.H{"status": "Success"})
+	go g.notifyGenericWebhookSimpleEvent(proj, payload, revision)
+	c.JSON(200, gin.H{"status": "Success. Build created"})
 }
 
-func (g *genericWebhook) notifyGenericWebhookEvent(proj *brigade.Project, payload []byte, gwData *genericWebhookData) {
-	if err := g.genericWebhookEvent(proj, payload, gwData); err != nil {
-		log.Printf("failed genericWebhook event: %s", err)
+func (g *genericWebhookSimpleEvent) notifyGenericWebhookSimpleEvent(proj *brigade.Project, payload []byte, revision *brigade.Revision) {
+	if err := g.genericWebhookSimpleEvent(proj, payload, revision); err != nil {
```

Comment:
Why the indirection here?

Reply:
This follows the pattern [here](https://github.com/Azure/brigade/blob/master/pkg/webhook/dockerhub.go#L68). Indeed, if we decide to make the call synchronous, there's no need for that.